### PR TITLE
266 feature request add exposure classes from nen en 1992 1 1

### LINKED ISSUE #266: [✨ Feature request]: Add exposure classes from NEN-EN 1992-1-1
### Description

- [x] Implement exposure classes

### Acceptance Criteria

- [x] Add the exposure classes to the code
- [x] Add unittests
- [x] Add docstrings

### Dependencies

_No response_

### Confirmation

- [x] I have checked that this feature does not already exist in the issue tracker.
- [x] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by SZeltaat**
## Description
Added definition of exposure classes + tests.

Fixes #266 

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/checks/concrete_cover/exposure_classes.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        Returns
+        -------
+        str
+            description of this exposure class
+        """
+        return "Corrosion induced by chlorides from sea water"
+
+
+@total_ordering
+class Freeze(Exposure):
```
Comment (PabloVasconez) [inline blueprints/checks/concrete_cover/exposure_classes.py:158]: Maybe rename to `FreezeThaw`

File: blueprints/checks/concrete_cover/exposure_classes.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    @staticmethod
+    def description() -> str:
+        """Static method which returns the description of this exposure class.
+
+        Returns
+        -------
+        str
+            description of this exposure class
+        """
+        return "Corrosion induced by carbonation"
```
Comment (bro-wi) [inline blueprints/checks/concrete_cover/exposure_classes.py:112]: I would like to add the description of the specific Enum and change this in another name like 'class_description'. Here s the list with all descriptions:

    EXPOSURE_CLASS_DESCRIPTION = {
        "Not applicable": "",
        "XC1": "Dry or permanently wet",
        "XC2": "Wet, rarely dry",
        "XC3": "Moderate humidity",
        "XC4": "Cyclic wet and dry",
        "XD1": "Moderate humidity",
        "XD2": "Wet, rarely dry",
        "XD3": "Cyclic wet and dry",
        "XS1": "Exposed to airborne salt but not in direct contact with sea water",
        "XS2": "Permanently submerged",
        "XS3": "Tidal, splash and spray zones",
        "XF1": "Moderate water saturation, without de-icing agent",
        "XF2": "Moderate water saturation, with de-icing agent",
        "XF3": "High water saturation, without de-icing agent",
        "XF4": "High water saturation, with de-icing agent or sea water",
        "XA1": "Slightly aggressive chemical environment",
        "XA2": "Moderately aggressive chemical environment",
        "XA3": "Highly aggressive chemical environment",
    }

**Review by bro-wi — COMMENTED** [review]
Nicely done! beautiful implementation.

File: blueprints/checks/concrete_cover/exposure_classes.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+        Returns
+        -------
+        str
+            String representation of the ExposureClasses object
+        """
+        return ", ".join(
+            enum.value
+            for enum in self
+            if enum.value != "Not applicable"  # pylint: disable=not-an-iterable
```
Comment (PabloVasconez) [inline blueprints/checks/concrete_cover/exposure_classes.py:220]: can be fix de pylint somehow?

**Review by PabloVasconez — APPROVED** [review]

File: blueprints/checks/concrete_cover/exposure_classes.py (hunk 3 above)
Comment (bro-wi) [reply blueprints/checks/concrete_cover/exposure_classes.py:220]: good catch, I think this is a bit of legacy code of us, we don't use pylint here ;-)

File: blueprints/checks/concrete_cover/exposure_classes.py [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+        """Description of subclasses to be implemented in each subclass.
+
+        Returns
+        -------
+        str
+            description of the specific exposure class
+        """
+        raise NotImplementedError("The description method must be implemented in the subclass!")
+
+    def description_of_the_environment(self) -> str:
```
Comment (PabloVasconez) [inline blueprints/checks/concrete_cover/exposure_classes.py:92]: maybe rename method to `environment_description`?

Comment (PabloVasconez) [reply blueprints/checks/concrete_cover/exposure_classes.py:92]: The name `description_of_the_environment` matches exactly the Eurocode.

**Review by egarciamendez — APPROVED** [review]
❤️

**Pull request merged**